frontend/pages/LabelPage/LabelPage.tsx

```diff
@@ -0,0 +1,205 @@
+  const isEditLabel = !location.pathname.includes("new");
+
+  const [selectedLabel, setSelectedLabel] = useState<ILabel>();
+  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(true);
+  const [
+    showOpenSchemaActionText,
+    setShowOpenSchemaActionText,
+  ] = useState<boolean>(false);
+  const [labelValidator, setLabelValidator] = useState<{
+    [key: string]: string;
+  }>(DEFAULT_CREATE_LABEL_ERRORS);
+  const { selectedOsqueryTable, setSelectedOsqueryTable } = useContext(
+    QueryContext
+  );
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const showSidebar = isSidebarOpen;
+
+  const { data: labels, error: labelsError } = useQuery<
+    ILabelsResponse,
+    Error,
+    ILabel[]
+  >(["labels"], () => labelsAPI.loadAll(), {
+    select: (data: ILabelsResponse) => data.labels,
+    onSuccess: (responseLabels: ILabel[]) => {
+      if (params.label_id) {
+        const selectLabel = responseLabels.find(
+          (label) => label.id === parseInt(params.label_id, 10)
+        );
+        setSelectedLabel(selectLabel);
+      }
+    },
+  });
+
+  useEffect(() => {
+    setShowOpenSchemaActionText(!isSidebarOpen);
+  }, [isSidebarOpen]);
+
+  const onCloseSchemaSidebar = () => {
+    setIsSidebarOpen(false);
+  };
+
+  const onOpenSchemaSidebar = () => {
+    setIsSidebarOpen(true);
+  };
+
+  const onOsqueryTableSelect = (tableName: string) => {
+    setSelectedOsqueryTable(tableName);
+  };
+
+  const onEditLabel = (formData: ILabelFormData) => {
+    if (!selectedLabel) {
+      console.error("Label isn't available. This should not happen.");
+      return;
+    }
+
+    const updateAttrs = deepDifference(formData, selectedLabel);
+
+    labelsAPI
+      .update(selectedLabel, updateAttrs)
+      .then(() => {
+        router.push(PATHS.MANAGE_HOSTS_LABEL(selectedLabel.id));
+        renderFlash(
+          "success",
+          "Label updated. Try refreshing this page in just a moment to see the updated host count for your label."
+        );
+        setLabelValidator({});
+      })
+      .catch((updateError: { data: IApiError }) => {
+        if (updateError.data.errors[0].reason.includes("Duplicate")) {
+          setLabelValidator({
+            name: "A label with this name already exists",
+          });
+        } else if (
+          updateError.data.errors[0].reason.includes(
+            "Data too long for column 'name'"
+          )
+        ) {
+          setLabelValidator({
+            name: "Label name is too long",
+          });
+        } else if (
+          updateError.data.errors[0].reason.includes(
+            "Data too long for column 'description'"
+          )
+        ) {
+          setLabelValidator({
+            description: "Label description is too long",
+          });
+        } else {
+          renderFlash("error", "Could not create label. Please try again.");
+        }
+      });
+  };
+
+  const onAddLabel = (formData: ILabelFormData) => {
```

Comment:
Let's chat a bit around these api calls within the components. Got some ideas and wanted to see what you thought about it.